File: web/misc/test/api/mw_api.py
# coding: utf-8
# +-----------------------------------------------------------------------------------
# | MW Linux面板
# +-----------------------------------------------------------------------------------
# | Copyright (c) 2015-2099 MW(http://github.com/midoks/mdserver) All rights reserved.
# +-----------------------------------------------------------------------------------
# | Author: midoks
# +-----------------------------------------------------------------------------------

#------------------------------
# API-Demo of Python
#------------------------------
import time
import hashlib
import sys
import os
import json


class mwApi:
    __MW_PANEL = 'http://154.12.53.90:51377/'
    __MW_APP_ID = 'yhYkxGssPD'
    __MW_APP_SERECT = 'ErmBdr563eJ5GMM5sWbc'
    
    # 如果希望多台面板，可以在实例化对象时，将面板地址与密钥传入
    def __init__(self, panel_url=None, app_id=None, app_serect=None):
        if panel_url:
            self.__MW_PANEL = panel_url
            self.__MW_APP_ID = app_id
            self.__MW_APP_SERECT = app_serect

    def post(self, endpoint, request_data):
        import requests
        url = self.__MW_PANEL + endpoint  
        post_data = requests.post(url, data=request_data, headers={
            'app-id':self.__MW_APP_ID,
            'app-secret':self.__MW_APP_SERECT
        })
        try:
            return post_data.json()
        except Exception as e:
            return post_data.text
    # 取面板日志
    def getLogs(self):
        result = self.post('/task/count',{'limit':10,'p':1})
        return result


if __name__ == '__main__':
    # 实例化MW-API对象
    api = mwApi()

    # 调用get_logs方法
    rdata = api.getLogs()

    # 打印响应数据
    print(rdata)

File: web/misc/test/api/mw_api.php
<?php
/**
 * MW API接口示例Demo
 * 仅供参考，请根据实际项目需求开发，并做好安全处理
 * date 2022-11-28
 * author midoks
 */
class mwApi {
	private $MW_PANEL      = "http://127.0.0.1:64307"; //面板地址
	private $MW_APP_ID     = 'hC6XArWzRY';
	private $MW_APP_SERECT = 'NSGaFhMWyaN5Yi3ftTkZ';

	//如果希望多台面板，可以在实例化对象时，将面板地址与密钥传入
	public function __construct($mw_panel = null, $app_id = null, $app_secret = null) {
		if ($mw_panel) {
			$this->MW_PANEL = $mw_panel;
		}

		if ($app_id) {
			$this->MW_APP_ID = $app_id;
		}

		if ($app_secret) {
			$this->MW_APP_SERECT = $app_secret;
		}
	}

	/**
	 * 发起POST请求
	 * @param String $url 目标网填，带http://
	 * @param Array|String $data 欲提交的数据
	 * @return string
	 */
	private function httpPost($url, $data, $timeout = 60) {

		$ch = curl_init();
		// 设置头部信息
		$headers = [
			'app-id: ' . $this->MW_APP_ID,
			'app-secret: ' . $this->MW_APP_SERECT,
		];
		curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
		curl_setopt($ch, CURLOPT_URL, $url);
		curl_setopt($ch, CURLOPT_TIMEOUT, $timeout);
		curl_setopt($ch, CURLOPT_POST, 1);
		curl_setopt($ch, CURLOPT_POSTFIELDS, $data);
		curl_setopt($ch, CURLOPT_COOKIEJAR, $cookie_file);
		curl_setopt($ch, CURLOPT_COOKIEFILE, $cookie_file);
		curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
		curl_setopt($ch, CURLOPT_HEADER, 0);
		curl_setopt($ch, CURLOPT_SSL_VERIFYHOST, false);
		curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);
		$output = curl_exec($ch);
		curl_close($ch);
		return $output;
	}

	public function panel($endpoint, $data) {
		$url = $this->MW_PANEL . $endpoint;
		return $this->httpPost($url, $data);
	}

	//示例取面板日志
	public function getLogsList() {
		$post_data['p']     = '1';
		$post_data['limit'] = 10;

		//请求面板接口
		$data = $this->panel('/logs/get_log_list', $post_data);

		//解析JSON数据
		// $data = json_decode($result, true);
		return $data;
	}

}

//实例化对象
$api = new mwApi();
//获取面板日志
$rdata = $api->getLogsList();

// var_dump($rdata);
//输出JSON数据到浏览器
echo json_encode($rdata);

?>